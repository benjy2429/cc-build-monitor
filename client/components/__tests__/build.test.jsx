import React from 'react';
import { shallow } from 'enzyme';
import Build from '../build';

const defaultProps = {
  reponame: 'my-project',
  branch: 'master',
  build_num: 123,
  status: 'success',
  build_time_millis: 5000,
  user: {
    name: 'Joe Bloggs',
    avatar_url: 'https://example.com/image.png',
  },
};

describe('Build', () => {
  it('renders correctly', () => {
    const component = shallow(
      <Build {...defaultProps} />,
    );

    expect(component).toMatchSnapshot();
  });

  it('displays the build time when running', () => {
    const props = {
      ...defaultProps,
      status: 'running',
    };
    const component = shallow(
      <Build {...props} />,
    );

    expect(component).toMatchSnapshot();
  });

  it('hides the time if it is not in the props', () => {
    const props = {
      ...defaultProps,
      build_time_millis: undefined,
    };
    const component = shallow(
      <Build {...props} />,
    );

    expect(component).toMatchSnapshot();
  });

  it('displays unknown if the status is does not have a friendly name', () => {
    const props = {
      ...defaultProps,
      status: 'something_else',
    };
    const component = shallow(
      <Build {...props} />,
    );

    expect(component).toMatchSnapshot();
  });
});
