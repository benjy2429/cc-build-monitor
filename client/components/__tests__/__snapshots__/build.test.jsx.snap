// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Build displays the build time when running 1`] = `
<div
  className="build-bar build-running"
>
  <div
    className="build-status"
  >
    Running
  </div>
  <img
    alt="Joe Bloggs"
    className="build-author-avatar"
    src="https://example.com/image.png"
    title="Joe Bloggs"
  />
  <div
    className="build-details"
  >
    <div
      className="build-name text-ellipsis"
    >
      #
      123
       - 
      my-project
       / 
      master
    </div>
  </div>
  <div
    className="build-time"
  >
    <Ticker
      ticking={true}
      time={5000}
    />
  </div>
</div>
`;

exports[`Build displays unknown if the status is does not have a friendly name 1`] = `
<div
  className="build-bar build-something_else"
>
  <div
    className="build-status"
  >
    Unknown
  </div>
  <img
    alt="Joe Bloggs"
    className="build-author-avatar"
    src="https://example.com/image.png"
    title="Joe Bloggs"
  />
  <div
    className="build-details"
  >
    <div
      className="build-name text-ellipsis"
    >
      #
      123
       - 
      my-project
       / 
      master
    </div>
  </div>
  <div
    className="build-time"
  >
    <Ticker
      ticking={false}
      time={5000}
    />
  </div>
</div>
`;

exports[`Build hides the time if it is not in the props 1`] = `
<div
  className="build-bar build-success"
>
  <div
    className="build-status"
  >
    Success
  </div>
  <img
    alt="Joe Bloggs"
    className="build-author-avatar"
    src="https://example.com/image.png"
    title="Joe Bloggs"
  />
  <div
    className="build-details"
  >
    <div
      className="build-name text-ellipsis"
    >
      #
      123
       - 
      my-project
       / 
      master
    </div>
  </div>
  <div
    className="build-time"
  />
</div>
`;

exports[`Build renders correctly 1`] = `
<div
  className="build-bar build-success"
>
  <div
    className="build-status"
  >
    Success
  </div>
  <img
    alt="Joe Bloggs"
    className="build-author-avatar"
    src="https://example.com/image.png"
    title="Joe Bloggs"
  />
  <div
    className="build-details"
  >
    <div
      className="build-name text-ellipsis"
    >
      #
      123
       - 
      my-project
       / 
      master
    </div>
  </div>
  <div
    className="build-time"
  >
    <Ticker
      ticking={false}
      time={5000}
    />
  </div>
</div>
`;
